src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -209,9 +214,34 @@ def evaluate_group_key_value(
                     new_substatus=None,
                 )
             else:
-                result, event_data = self.build_occurrence_and_event_data(
-                    group_key, value, PriorityLevel(new_status)
+                detector_occurrence, event_data = self.build_occurrence_and_event_data(
+                    group_key, PriorityLevel(new_status)
                 )
+
+                evidence_data = {
+                    **detector_occurrence.evidence_data,
+                    "detector_id": self.detector.id,
+                    "value": value,
+                }
+
+                result = IssueOccurrence(
+                    str(uuid4()),
+                    self.detector.project_id,
+                    str(uuid4()),
+                    self.build_fingerprint(group_key),
```

Comment:
`uptime` and any other detectors that already use the issue platform will need the ability to produce their own values for fingerprint. To avoid making duplicate issues, initially we'll pass both the old and new fingerprint to the issue platform so they can be joined.

I guess this can be handled via overriding `build_fingerprint`

Reply:
🤔 yeah, i was more thinking about that use case (overriding build_fingerprint) -- if we're exposing it as a method already. I'd be okay with us _not_ exposing that method, and instead have a default value using an internal method and have it be passed in.